quinn-proto/src/connection.rs

```diff
@@ -1101,10 +1074,9 @@ impl Connection {
                                 self.handshake_cleanup(&ctx.config);
                                 let mut msgs = Vec::new();
                                 self.tls.write_tls(&mut msgs).unwrap();
-                                if self.side == Side::Client {
-                                    self.transmit_handshake(&msgs);
-                                } else {
-                                    self.transmit(StreamId(0), msgs.into());
+                                self.transmit_handshake(&msgs);
+                                if self.side == Side::Server {
+                                    self.awaiting_handshake = false;
```

Comment:
Why do we actually need `awaiting_handshake`? Doesn't that somehow duplicate other bits of state we have in the `Connection` (like, perhaps, the actual `state`)?

Reply:
It's a hack to partially mitigate the brokenness of the draft-11 handshake logic, specifically relating to race conditions around encryption level changes. Will be removed as part of the transition to the new handshake logic.

Reply:
Ok!